lib/Conversion/TorchToLinalg/Linear.cpp

```diff
@@ -779,6 +813,260 @@ class ConvertAtenBmmOp : public OpConversionPattern<AtenBmmOp> {
+                  indices[flipDim] = b.create<arith::SubIOp>(
+                      loc,
+                      b.create<arith::SubIOp>(loc, weightInitDims[flipDim], c1),
+                      indices[flipDim]);
+                }
+                Value res = b.create<tensor::ExtractOp>(loc, weight, indices)
+                                .getResult();
+                b.create<linalg::YieldOp>(loc, res);
+              })
+          .getResult(0);
+
+  // Calculate padded input size, allocate tensor
+  SmallVector<Value> outerSizes{inputBatch, inputChannels};
+  SmallVector<Value> innerSizes{inputBatch, inputChannels};
+  SmallVector<Value> offsets{c0, c0};
+  for (size_t i = 0; i < numSpatialDims; i++) {
+    Value innerSize =
+        rewriter.create<arith::SubIOp>(loc, inputSpatialDimensions[i], c1);
+    innerSize = rewriter.create<arith::MulIOp>(
+        loc, innerSize,
+        castIntToIndex(rewriter, loc, convolutionAttributes.strideValues[i]));
+    innerSize = rewriter.create<arith::AddIOp>(loc, innerSize, c1);
+
+    Value offset = rewriter.create<arith::SubIOp>(loc, weightDims[i], c1);
+    offset = rewriter.create<arith::MulIOp>(
+        loc, offset,
+        castIntToIndex(rewriter, loc, convolutionAttributes.dilationValues[i]));
+    offset = rewriter.create<arith::SubIOp>(
+        loc, offset,
+        castIntToIndex(rewriter, loc, convolutionAttributes.padding[i]));
+
+    Value outerSize = rewriter.create<arith::MulIOp>(loc, offset, c2);
+    outerSize = rewriter.create<arith::AddIOp>(loc, outerSize, innerSize);
+    outerSize = rewriter.create<arith::AddIOp>(
+        loc, outerSize,
+        castIntToIndex(rewriter, loc, convolutionAttributes.outputPadding[i]));
+
+    outerSizes.push_back(outerSize);
+    offsets.push_back(offset);
+  }
+
+  // Allocate padded input tensor
+  Value initTensor = createInitTensor(rewriter, loc, outerSizes, inputDTy, pad);
+
+  // Insert input into allocated tensor
+  SmallVector<Value> strideIndexValues{c1, c1};
+  for (auto stride : convolutionAttributes.strideValues)
+    strideIndexValues.push_back(castIntToIndex(rewriter, loc, stride));
+  SmallVector<Value> insertSizes = getTensorSizes(rewriter, loc, input);
+
+  paddedInput = rewriter.create<tensor::InsertSliceOp>(
+      loc, torch_to_linalg::removeSizeInformation(rewriter, loc, input),
+      initTensor, offsets, insertSizes, strideIndexValues);
+
+  // Calculate output dims
+  for (size_t i = 0; i < numSpatialDims; i++)
+    outDims.push_back(torch_to_linalg::getOutputDimForConvTransposeOps(
+        rewriter, loc, inputSpatialDimensions[i],
+        convolutionAttributes.padding[i],
+        convolutionAttributes.dilationValues[i],
+        rewriter.createOrFold<arith::IndexCastOp>(
+            loc, rewriter.getIntegerType(64), weightDims[i]),
+        convolutionAttributes.strideValues[i],
+        convolutionAttributes.outputPadding[i]));
+
+  // Set stride to 1
+  convolutionAttributes.stride.clear();
```

Comment:
Why is the stride cleared and simply set to 1?

Reply:
It is handled by the InsertSliceOp in line 1050. Not sure if this is the most performant way, but I basically just copied code and moved it into a function.